### PR TITLE
better error handling for signup route

### LINKED ISSUE #90: If server returns a statusCode:400, user still gets a "thank you" message when signing up to waitlist
Ref: https://github.com/mozilla/idea-town/pull/75#discussion_r36571105

```sh
$ git log -1 --oneline # 56bd040 Merge pull request #83 from mozilla/issue-70
```

### Steps to reproduce:
1. Set up idea-town locally (but don't configure EBS or AWS creds).
2. Sign up to the waitlist (you can use "foo@localhost").
3. Click the "Sign up" button to register for the waiting list.
4. View the Terminal console output.

### Actual results:
Since I don't have a backend set up, I'd expect a hard fail. But the front end shows a happy path while the server cries on the floor.

idea_town

1__node_server__node__and_issues_ _mozilla_idea-town_and_idea_town

```
Error adding item to database:  { [ResourceNotFoundException: Requested resource not found]
  message: 'Requested resource not found',
  code: 'ResourceNotFoundException',
  time: Tue Aug 11 2015 13:45:48 GMT-0700 (PDT),
  statusCode: 400,
  retryable: false,
  retryDelay: 0 }
```

### PR DESCRIPTION
**Pull request by meandavejustice**
Fixes #90

### CODE REVIEW

File: routes/signup.js [hunk 1]
```diff
@@ -7,12 +7,11 @@ module.exports = function(req, res) {
     var busboy = new Busboy({headers: req.headers});
     busboy.on('field', function(fieldname, val) {
       if (fieldname === 'email') {
```
Comment (6a68) [inline routes/signup.js]: @meandavejustice Seems like we should return a 400 if the email is invalid here

Comment (6a68) [inline routes/signup.js]: sorry - if the email is invalid or missing here

File: routes/signup.js [hunk 2]
```diff
@@ -29,9 +28,18 @@ function signup(emailSubmitted) {
...
     } else {
-      console.log('Form data added to database.');
+      sendSuccess(req, res);
     }
   });
 }
+
+function sendErr(req, res, err) {
+  console.error(err);
+  res.sendStatus(500)
```
Comment (6a68) [inline routes/signup.js]: derp, missing semicolon will probably make the linter cry

File: routes/signup.js [hunk 3]
```diff
@@ -20,7 +19,7 @@ module.exports = function(req, res) {
   }
```
Comment (6a68) [inline routes/signup.js]: should probably replace this `res.sendStatus` with a `sendErr` call, too?

File: routes/signup.js [hunk 4]
```diff
@@ -7,12 +7,11 @@ module.exports = function(req, res) {
...
-        signup(val);
+        signup(req, res, val);
       }
-    });
-    busboy.on('finish', function() {
-      res.sendStatus(200);
-    });
+    }).on('error', function(err) {
+      sendErr(req, res, err);
+    })
```
Comment (pdehaan) [inline routes/signup.js]: Missing semi.

```
[22:22:39] Using gulpfile ~/build/mozilla/idea-town/gulpfile.js
[22:22:39] Starting 'lint'...
[22:22:39] 'lint' errored after 309 ms
[22:22:39] ESLintError in plugin 'gulp-eslint'
Message:
    Missing semicolon.
Details:
    fileName: /home/travis/build/mozilla/idea-town/routes/signup.js
    lineNumber: 14
[22:22:39] 
/home/travis/build/mozilla/idea-town/routes/signup.js
  14:7   error  Missing semicolon  semi
  40:22  error  Missing semicolon  semi
✖ 2 problems (2 errors, 0 warnings)
```

File: routes/signup.js (hunk 2 above)
Comment (pdehaan) [inline routes/signup.js]: Missing semi.

```
[22:22:39] Using gulpfile ~/build/mozilla/idea-town/gulpfile.js
[22:22:39] Starting 'lint'...
[22:22:39] 'lint' errored after 309 ms
[22:22:39] ESLintError in plugin 'gulp-eslint'
Message:
    Missing semicolon.
Details:
    fileName: /home/travis/build/mozilla/idea-town/routes/signup.js
    lineNumber: 14
[22:22:39] 
/home/travis/build/mozilla/idea-town/routes/signup.js
  14:7   error  Missing semicolon  semi
  40:22  error  Missing semicolon  semi
✖ 2 problems (2 errors, 0 warnings)
```

File: routes/signup.js (hunk 3 above)
Comment (6a68) [inline routes/signup.js]: (this seems confusing - I was referring to the `res.sendStatus(500)` on line 18, but the diff tool wouldn't let me insert a comment there)

**Comment by 6a68** [comment]
@meandavejustice I'm going to touch these code paths to add basket integration, so I'm going to merge instead of waiting for the last two things to get fixed :tada:

**Pull request merged**